### DISCUSSION TITLE
Embedding quasar app in existing site

### DISCUSSION THREAD

**Discussion by ghost** [quasarframework/quasar#11632] [General - Components / Directives / etc] [answered]
Hello all,

First of all, thanks for making Quasar one of the most productive web frameworks in the market!
I have to embed a Quasar app in  an existing site and I would like to know if there are any best practices / recommendations on how to do that.
We have tried to simply add the quasar app using a script tag to load the app  from a remote location but we are experiencing some CSS interference between the existing site and the app. Some of the app's CSS "leaks" to the surrounding page and vice-versa. Maybe we did something wrong.

We can use iframe to better isolate the 2 apps but iframe is shunned by the security people involved.

Any ideas?

Thanks in advance!

**Comment by TobyMosque** [comment, marked as answer]
So, if u're trying to include the Quasar Framework in a old project (probably a MPA) who adopts another frontend framework (I'm looking to you Bootstrap) or even a proprietary design language who didn't follow the BEM 101, u'll be bound to the UMD version, and resolve the CSS conflicts by yourself.

I already did in the past, and i know how painful is to maintain them... but that isn't a Quasar fault, quasar itselft adopts BEM 101 in some degree, so quasar works well with other frameworks who also adopts BEM 101 (probably u'll face some conflicts, but will be few to none).

But if u're trying to integrate a Quasar App (SPA) in a old project (MPA), i recommend u to pick the [Vue Cli Flavor](https://quasar.dev/start/vue-cli-plugin) and bundle/target your App as a [Web Component](https://cli.vuejs.org/guide/build-targets.html#web-component)... doing that, you app will act as a microfrontend, so u'll be able to embedded that anywhere

**Reply by mila76** [reply]
I used the vue cli and build as library (in production used the UMD and "that library"). All work good very good in an old mpa.
I used this approach so I can initialize that part when I need and easily pass some data in the php template. 

But as @TobyMosque mentioned you need fix css leaking. I need to fix only few (because of some global resetting in forms)... Is not good as 100% but good enough for the client. (remain small problem but you can spot only if you know normal quasar appearance)
